app/src/app/utils/helpers.ts

```diff
@@ -119,64 +116,13 @@ export const getFeaturesIntersectingCounties = (
   });
 
   if (!countyFeatures) return;
+  const fips = countyFeatures[0].properties.STATEFP + countyFeatures[0].properties.COUNTYFP;
 
-  const featureBbox = getBoundingBoxFromFeatures(countyFeatures);
-
-  if (!featureBbox) return;
-
-  const sw = map.project(featureBbox[0]);
-  const ne = map.project(featureBbox[1]);
-  const features = map.queryRenderedFeatures([sw, ne], {
+  const features = map.queryRenderedFeatures(undefined, {
     layers,
   });
-
-  let countyPoly;
-  try {
-    // @ts-ignore: Property 'coordinates' does not exist on type 'Geometry'.
-    countyPoly = polygon(countyFeatures[0].geometry.coordinates);
-  } catch {
-    // @ts-ignore: Property 'coordinates' does not exist on type 'Geometry'.
-    countyPoly = multiPolygon(countyFeatures[0].geometry.coordinates);
-  }
-
-  return features.filter(p => {
-    const point = pointOnFeature(p);
-    return booleanWithin(point, countyPoly);
-  });
-};
-
-/**
- * getBoundingBoxFromCounties
- * Calculate the bounding box (SW and NE corners) from county features.
- * @param countyFeatures - Array of GeoJSON Features representing counties
- * @returns [PointLike, PointLike] - An array containing the SW and NE corners of the bounding box
- */
-const getBoundingBoxFromFeatures = (
-  features: MapGeoJSONFeature[]
-): [LngLatLike, LngLatLike] | null => {
-  if (!features || features.length === 0) {
-    return null;
-  }
-
-  const sw = new LngLat(180, 90);
-  const ne = new LngLat(-180, -90);
-
-  features.forEach(feature => {
-    // this will always have an even number of coordinates
-    // iterating over the coordinates in pairs yields (lng, lat)
-    // @ts-ignore: Property 'coordinates' does not exist on type 'Geometry'.
-    let coords = feature.geometry.coordinates.flat(Infinity);
-    for (let i = 0; i < coords.length; i += 2) {
-      let x = coords[i];
-      let y = coords[i + 1];
-      sw.lng = Math.min(sw.lng, x);
-      sw.lat = Math.min(sw.lat, y);
-      ne.lng = Math.max(ne.lng, x);
-      ne.lat = Math.max(ne.lat, y);
-    }
-  });
-
-  return [sw, ne];
+  return features.filter(p => (p?.id) &&
```

Comment:
Q: This approach won't capture geometries that are not rendered. We've discussed that the issue of non-rendered geometries is not too bad for painting blocks, but in this case a county could extend outside of the window bounds.

Should we consider an approach with a backend query to get all the expected IDs?

Eg.pseudo
```javascript
const fips = brushedFeature.properties.fips
const countyChildren = await fetch(`api/county_children/{document_id}/{fips}`)
setZoneAssignments(
  countyChildren,
  zone
)
```

Reply:
PP: it's reasonable that a user might only paint what they see. Since the current functionality is actively buggy I'd recommend we go with this and can create an issue to experiment w/ a backend approach. Also–and I'm not wedded to this opinion–there's something kind of nice about FE being responsible for painting and API/DB only being responsible for documents and their assignments

Reply:
Sounds good!